clusterloader2/pkg/measurement/util/propagation_metric.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/montanaflynn/stats"
+	"github.com/prometheus/common/model"
+	"k8s.io/klog/v2"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	defaultDNSPropagationProbeNamespaceIndex = 0
+	defaultDNSPropagationProbeSampleCount    = 10
+)
+
+// NewPropagationMetricPrometheus tries to parse latency data from results of Prometheus query from propagation probe.
+func NewPropagationMetricPrometheus(samples []*model.Sample) (*LatencyMetric, error) {
+	var latencyMetric LatencyMetric
+	values := []float64{}
+	for _, sample := range samples {
+		values = append(values, float64(sample.Value))
+	}
+	sort.Float64s(values)
+
+	p50, _ := stats.Percentile(values, 50)
+	p90, _ := stats.Percentile(values, 90)
+	p99, _ := stats.Percentile(values, 99)
+
+	latencyMetric.SetQuantile(0.5, time.Duration(p50*float64(time.Second)))
+	latencyMetric.SetQuantile(0.90, time.Duration(p90*float64(time.Second)))
+	latencyMetric.SetQuantile(0.99, time.Duration(p99*float64(time.Second)))
+	return &latencyMetric, nil
+}
+
+func InitializeTemplateMappingForDNSPropagationProbe(config *measurement.Config) map[string]interface{} {
+	DNSPropagationProbeService, err := util.GetString(config.Params, "DNSPropagationProbeService")
+	if err != nil {
+		DNSPropagationProbeService = ""
+	}
+	DNSPropagationProbeStatefulSet, err := util.GetString(config.Params, "DNSPropagationProbeStatefulSet")
+	if err != nil {
+		DNSPropagationProbeStatefulSet = ""
+	}
+	DNSPropagationProbePodCount, err := util.GetInt(config.Params, "DNSPropagationProbePodCount")
+	if err != nil {
+		DNSPropagationProbePodCount = 0
+	}
+
+	DNSPropagationProbeSampleCount, err := util.GetInt(config.Params, "DNSPropagationProbeSampleCount")
```

Comment:
Nit: `GetIntOrDefault`?

Dittos for the above getters.

Reply:
Done